include/c_dataset.h

```diff
@@ -49,6 +50,13 @@ extern "C" {
 void* CDataSet(const char* name,
                const size_t name_length);
 
+/*!
+*   \brief C-DataSet destructor
+*   \param name The name of the dataset
+*   \param dataset A c-ptr to the dataset object
+*/
+void DeallocateeDataSet(void* dataset);
```

Comment:
In the  c client interface we use ``DeleteCClient``.  Should we call this ``DeleteDataSet`` to be consistent?

Reply:
There is already a delete_dataset function that removes the dataset from Redis, so I wanted to try to avoid confusion